### ISSUE TITLE
Fail on android build

### ISSUE THREAD

**Issue by NCrashed** [hexresearch/ergvein#126]
Branch `wallet-storage`, build log:

```
walk_tree.c:104:43: warning: suggest parentheses around '&&' within '||' [-Wparentheses]
         if ((flags & WALK_TREE_RECURSIVE) &&
                                           ^
/nix/store/b9p787yqaqi313l9rr0491igjwyzqfmw-bash-4.4-p23/bin/bash ../libtool --mode=link aarch64-unknown-linux-android-gcc  -o libmisc.la quote.lo unquote.lo high_water_alloc.lo next_line.lo walk_tree.lo 
libtool: link: aarch64-unknown-linux-android-ar cru .libs/libmisc.a .libs/quote.o .libs/unquote.o .libs/high_water_alloc.o .libs/next_line.o .libs/walk_tree.o 
aarch64-unknown-linux-android-ar: libtinfo.so.5: cannot open shared object file: No such file or directory

libtool: link: aarch64-unknown-linux-android-ranlib .libs/libmisc.a
aarch64-unknown-linux-android-ranlib: libtinfo.so.5: cannot open shared object file: No such file or directory

libtool: link: ( cd ".libs" && rm -f "libmisc.la" && ln -s "../libmisc.la" "libmisc.la" )
make[2]: Leaving directory '/build/attr-2.4.47/libmisc'
/nix/store/8m2ld502dsx6rbsvv05597qzxha4cnc1-gnumake-4.2.1/bin/make -C libattr
make[2]: Entering directory '/build/attr-2.4.47/libattr'
/nix/store/b9p787yqaqi313l9rr0491igjwyzqfmw-bash-4.4-p23/bin/bash ../libtool --mode=compile aarch64-unknown-linux-android-gcc -g -O2 -DDEBUG -funsigned-char -fno-strict-aliasing -Wall -DVERSION=\"2.4.47\" -DLOCALEDIR=\"/nix/store/dlrviakg03h93m8fyblf6hdvaf6f9bcy-attr-2.4.47-aarch64-unknown-linux-android/share/locale\" -DPACKAGE=\"attr\" -I../include -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -include libattr.h  -c libattr.c
libtool: compile:  aarch64-unknown-linux-android-gcc -g -O2 -DDEBUG -funsigned-char -fno-strict-aliasing -Wall -DVERSION=\"2.4.47\" -DLOCALEDIR=\"/nix/store/dlrviakg03h93m8fyblf6hdvaf6f9bcy-attr-2.4.47-aarch64-unknown-linux-android/share/locale\" -DPACKAGE=\"attr\" -I../include -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -include libattr.h -c libattr.c  -fPIC -DPIC -o .libs/libattr.o
In file included from libattr.c:25:0:
../include/attr/xattr.h: In function 'setxattr':
../include/attr/xattr.h:37:58: error: expected declaration specifiers before '__THROW'
         const void *__value, size_t __size, int __flags) __THROW;
                                                          ^
../include/attr/xattr.h:39:58: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__THROW'
         const void *__value, size_t __size, int __flags) __THROW;
                                                          ^
../include/attr/xattr.h:41:58: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__THROW'
         const void *__value, size_t __size, int __flags) __THROW;
                                                          ^
../include/attr/xattr.h:44:35: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__THROW'
     void *__value, size_t __size) __THROW;
                                   ^
../include/attr/xattr.h:46:35: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__THROW'
     void *__value, size_t __size) __THROW;
                                   ^
../include/attr/xattr.h:48:35: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__THROW'
     void *__value, size_t __size) __THROW;
                                   ^
../include/attr/xattr.h:51:20: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__THROW'
     size_t __size) __THROW;
                    ^
../include/attr/xattr.h:53:20: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__THROW'
     size_t __size) __THROW;
                    ^
../include/attr/xattr.h:55:20: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__THROW'
     size_t __size) __THROW;
                    ^
../include/attr/xattr.h:57:65: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__THROW'
 extern int removexattr (const char *__path, const char *__name) __THROW;
                                                                 ^
../include/attr/xattr.h:58:66: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__THROW'
 extern int lremovexattr (const char *__path, const char *__name) __THROW;
                                                                  ^
../include/attr/xattr.h:59:63: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__THROW'
 extern int fremovexattr (int __filedes,   const char *__name) __THROW;
                                                               ^
In file included from libattr.c:26:0:
../include/attr/attributes.h:65:3: error: storage class specified for parameter 'attrlist_t'
 } attrlist_t;
   ^
../include/attr/attributes.h:74:3: error: storage class specified for parameter 'attrlist_ent_t'
 } attrlist_ent_t;
   ^
../include/attr/attributes.h:94:3: error: storage class specified for parameter 'attrlist_cursor_t'
 } attrlist_cursor_t;
   ^
../include/attr/attributes.h:106:3: error: storage class specified for parameter 'attr_multiop_t'
 } attr_multiop_t;
   ^
../include/attr/attributes.h:122:12: error: storage class specified for parameter 'attr_get'
 extern int attr_get (const char *__path, const char *__attrname,
            ^
../include/attr/attributes.h:124:12: error: storage class specified for parameter 'attr_getf'
 extern int attr_getf (int __fd, const char *__attrname, char *__attrvalue,
            ^
../include/attr/attributes.h:131:12: error: storage class specified for parameter 'attr_set'
 extern int attr_set (const char *__path, const char *__attrname,
            ^
../include/attr/attributes.h:134:12: error: storage class specified for parameter 'attr_setf'
 extern int attr_setf (int __fd, const char *__attrname,
            ^
../include/attr/attributes.h:142:12: error: storage class specified for parameter 'attr_remove'
 extern int attr_remove (const char *__path, const char *__attrname,
            ^
../include/attr/attributes.h:144:12: error: storage class specified for parameter 'attr_removef'
 extern int attr_removef (int __fd, const char *__attrname, int __flags);
            ^
../include/attr/attributes.h:154:16: error: expected declaration specifiers or '...' before 'attrlist_cursor_t'
   int __flags, attrlist_cursor_t *__cursor);
                ^
../include/attr/attributes.h:156:16: error: expected declaration specifiers or '...' before 'attrlist_cursor_t'
   int __flags, attrlist_cursor_t *__cursor);
                ^
../include/attr/attributes.h:175:44: error: expected declaration specifiers or '...' before 'attr_multiop_t'
 extern int attr_multi (const char *__path, attr_multiop_t *__oplist,
                                            ^
../include/attr/attributes.h:177:35: error: expected declaration specifiers or '...' before 'attr_multiop_t'
 extern int attr_multif (int __fd, attr_multiop_t *__oplist,
                                   ^
libattr.c:36:20: error: storage class specified for parameter 'user_name'
 static const char *user_name = "user.";
                    ^
libattr.c:36:1: error: parameter 'user_name' is initialized
 static const char *user_name = "user.";
 ^
libattr.c:37:20: error: storage class specified for parameter 'secure_name'
 static const char *secure_name = "security.";
                    ^
libattr.c:37:1: error: parameter 'secure_name' is initialized
 static const char *secure_name = "security.";
 ^
libattr.c:38:20: error: storage class specified for parameter 'trusted_name'
 static const char *trusted_name = "trusted.";
                    ^
libattr.c:38:1: error: parameter 'trusted_name' is initialized
 static const char *trusted_name = "trusted.";
 ^
libattr.c:39:20: error: storage class specified for parameter 'xfsroot_name'
 static const char *xfsroot_name = "xfsroot.";
                    ^
libattr.c:39:1: error: parameter 'xfsroot_name' is initialized
 static const char *xfsroot_name = "xfsroot.";
 ^
libattr.c:47:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
 {
 ^
libattr.c:68:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
 {
 ^
libattr.c:106:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
 {
 ^
libattr.c:130:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
 {
 ^
libattr.c:151:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
 {
 ^
libattr.c:178:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
 {
 ^
libattr.c:201:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
 {
 ^
libattr.c:221:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
 {
 ^
libattr.c:245:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
 {
 ^
libattr.c:268:4: error: expected declaration specifiers or '...' before 'attrlist_cursor_t'
    attrlist_cursor_t *cursor)
    ^
libattr.c:314:5: error: expected declaration specifiers or '...' before 'attrlist_cursor_t'
     attrlist_cursor_t *cursor)
     ^
libattr.c:360:31: error: expected declaration specifiers or '...' before 'attr_multiop_t'
 attr_single(const char *path, attr_multiop_t *op, int flags)
                               ^
libattr.c:378:28: error: expected declaration specifiers or '...' before 'attr_multiop_t'
 attr_singlef(const int fd, attr_multiop_t *op, int flags)
                            ^
libattr.c:403:30: error: expected declaration specifiers or '...' before 'attr_multiop_t'
 attr_multi(const char *path, attr_multiop_t *multiops, int count, int flags)
                              ^
libattr.c:420:21: error: expected declaration specifiers or '...' before 'attr_multiop_t'
 attr_multif(int fd, attr_multiop_t *multiops, int count, int flags)
                     ^
In file included from libattr.c:25:0:
../include/attr/xattr.h:36:12: error: old-style parameter declarations in prototyped function definition
 extern int setxattr (const char *__path, const char *__name,
            ^
libattr.c:434:1: error: expected '{' at end of input
 }
 ^
libattr.c:434:1: warning: control reaches end of non-void function [-Wreturn-type]
make[2]: *** [../include/buildrules:52: libattr.lo] Error 1
make[2]: Leaving directory '/build/attr-2.4.47/libattr'
make[1]: *** [include/buildrules:35: libattr] Error 2
make[1]: Leaving directory '/build/attr-2.4.47'
make: *** [Makefile:47: default] Error 2
builder for '/nix/store/z77rvi40w2s7mihmz4x2g2f7lsg00m8i-attr-2.4.47-aarch64-unknown-linux-android.drv' failed with exit code 2
building '/nix/store/9x9lvni28faxv9kis62yxzxmgpl8yi2g-attr-2.4.47-armv7a-unknown-linux-androideabi.drv'...
cannot build derivation '/nix/store/d1rshnvxfb9cms6nv4ypn52nwlqm8dsv-coreutils-8.29-aarch64-unknown-linux-android.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/hid270ljndmrc6zlpb1sbhwgs3flq9ra-git-2.18.1-aarch64-unknown-linux-android.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/ay5dklcvjas5gx42idhkmlkl3nlf2594-servant-reflex-9310745.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/ya3rxy08ppw7vzmxngjlgh9slxklckda-servant-reflex-0.3.4-aarch64-unknown-linux-android.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/bldmb6hdprbqb7gjkvz51f6nfy1v663v-ergvein-wallet-0.1.0.0-aarch64-unknown-linux-android.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/xccdmrwm2v2dcimnz0dl7p7wywn7148i-android-app.drv': 1 dependencies couldn't be built
building '/nix/store/n4fhyrz80qyfgm22g2gxza6nvlmichxn-auto-parent-3.drv'...
cannot build derivation '/nix/store/divjlcn2r4ykvabqb6p08v4fxfadidyl-org.ergvein.wallet.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/hh5flpfs1lcfax87397lpi5kx9bgj8yi-android-app.drv': 1 dependencies couldn't be built
error: build of '/nix/store/hh5flpfs1lcfax87397lpi5kx9bgj8yi-android-app.drv' failed
```

**Comment by NCrashed** [comment]
Need to get rid of servant-reflex.

**Comment by NCrashed** [comment]
Fixed in #129
